### PR TITLE
issue-1174

### PR DESCRIPTION
**Pull request by katarzyna-kulpa**
### Resolves #

_Describe your changes_

## PR checklist
- [ ] Add link to the issue
- [ ] Choose Project
- [ ] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: pkg/node/volumemgr_test.go [hunk 1]
```diff
@@ -2125,3 +2125,45 @@ func TestVolumeManager_failedToRemoveFinalizer(t *testing.T) {
 	_, err = vm.removeFinalizer(context.TODO(), volumeCR)
 	assert.NotNil(t, err)
 }
+
+func TestVolumeManager_retryDriveUpdate(t *testing.T) {
```
Comment (rafalw82) [inline pkg/node/volumemgr_test.go:2129]: Can we also check via expectEvent if the event was raised?

File: pkg/node/volumemgr.go [hunk 2]
```diff
@@ -376,18 +379,44 @@ func (m *VolumeManager) updateVolumeAndDriveUsageStatus(ctx context.Context, vol
...
+
+func (m *VolumeManager) retryDriveUpdate(ctx context.Context, volume *volumecrd.Volume, drive *drivecrd.Drive, driveStatus string) error {
+	ll := m.log.WithFields(logrus.Fields{
+		"method":      "retryDriveUpdate",
+		"volumeID":    volume.Name,
+		"driveStatus": driveStatus,
+	})
+	driveRemovalFailedEventSent := false
+	for i := 0; i < numberOfRetries; i++ {
+		if driveStatus == apiV1.DriveUsageFailed && !driveRemovalFailedEventSent {
```
[the remark this answers is not in the document]
Comment (katarzyna-kulpa) [reply pkg/node/volumemgr.go:398]: right it is not needed, thanks!

[the remark this answers is not in the document]
Comment (katarzyna-kulpa) [reply pkg/node/volumemgr.go]: done

File: pkg/node/volumemgr.go [hunk 3]
```diff
@@ -376,18 +379,44 @@ func (m *VolumeManager) updateVolumeAndDriveUsageStatus(ctx context.Context, vol
...
 			m.recorder.Eventf(drive, eventing.DriveRemovalFailed, eventMsg)
+			driveRemovalFailedEventSent = true
 		}
+
 		drive.Spec.Usage = driveStatus
 		if err := m.k8sClient.UpdateCR(ctx, drive); err != nil {
-			ll.Errorf("Unable to change drive %s usage status to %s, error: %v.",
-				drive.Name, drive.Spec.Usage, err)
-			return ctrl.Result{Requeue: true}, err
+			ll.Infof("Retrying to update drive %s usage status to %s. Sleep %d seconds and retry ...",
```
[the remark this answers is not in the document]
Comment (katarzyna-kulpa) [reply pkg/node/volumemgr.go]: done

File: pkg/node/volumemgr_test.go (hunk 1 above)
Comment (katarzyna-kulpa) [reply pkg/node/volumemgr_test.go:2129]: done

**Review by rafalw82 — APPROVED** [review]

**Review by makowalcz — APPROVED** [review]

**Pull request merged**